src/Sales.cpp

```diff
@@ -0,0 +1,118 @@
+#include "Sales.hpp"
+
+/*
+ * Constructor for base Sales class, initializes data to given values.
+ */
+Sale::Sale(const unsigned int id, const unsigned int sn, const std::string& d, const unsigned int as, const double sp, const double t, const std::string& b, const std::string& s)
+: identification(id), sale_number(sn), date(d), amount_sold(as), sale_price(sp), tax(t), buyer(b), seller(s)
+{
+    total_price = sale_price + (tax * sale_price);
+}
+
+Sale::~Sale()
+{
+    /*
+     * Shared pointers are used, so pointers will free themselves.
+     */
+}
+
+/*
+ * Initilizes what file will be used to load/save the sales data to.
+ */
+SaleList::SaleList(const std::string& f)
+{
+    file_name = f;
+    offset = 0;
+}
+
+/*
+ * Creates a new sale to put in the sales vector.
+ * May change to throwing an exeption in the futer.
+ * Also may change error checking here based on Sales date restrictions.
+ */
+bool SaleList::addSale(const unsigned int id, const unsigned int sn, const std::string& d, const unsigned int as, const double sp, const double t, const std::string& b, const std::string& s)
+{
+    if(id == 0 || sn == 0 || as == 0 || sp == 0 || d == "" || b == "") 
+    {
+        std::cerr << "Failed to read Sales Input. Note that the ID, Sales Number, and Sales Price cannot be 0 and a Date and Buyer must be given.\nContinuing to read."
```

Comment:
Also, this line is missing a semicolon at the end.  Also, go ahead and do ```<< std::endl;``` following your string just to make sure a newline is output and the output buffer is flushed.

I've forgotten a couple of times in the past, but please try to run ```make clean && make all``` before pushing up code to make sure that it compiles; especially for the duration that we don't have a CI build on PRs, as otherwise, we'd be merging broken code.